WcaOnRails/app/models/competition.rb

```diff
@@ -483,7 +483,7 @@ def user_can_view?(user)
   end
 
   def user_can_view_results?(user)
-    results_posted? || (user && user.can_admin_results?)
+    results_posted? || (!results.empty? && user && user.can_admin_results?)
```

Comment:
From a performance perspective, I think it would actually be better to reorder the conditions here a bit:

```ruby
results_posted? || (user && user.can_admin_results? && !results.empty?)
```

That way we don't bother to check `results.empty?` until we really have to.